pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalanceProgressStats.java

```diff
@@ -127,4 +163,77 @@ public static boolean statsDiffer(RebalanceStateStats base, RebalanceStateStats
     }
     return false;
   }
+
+  // TODO: Clean this up once new stats are verified
+  public static class RebalanceStateStats {
+    public int _segmentsMissing;
+    public int _segmentsToRebalance;
+    public double _percentSegmentsToRebalance;
+    public int _replicasToRebalance;
+
+    RebalanceStateStats() {
+      _segmentsMissing = 0;
+      _segmentsToRebalance = 0;
+      _replicasToRebalance = 0;
+      _percentSegmentsToRebalance = 0.0;
+    }
+  }
+
+  // These rebalance stats specifically track the total segments added / deleted across all replicas
+  public static class RebalanceProgressStats {
+    // Total segments - across all replicas
+    @JsonProperty("totalSegmentsToBeAdded")
+    public int _totalSegmentsToBeAdded;
+    @JsonProperty("totalSegmentsToBeDeleted")
+    public int _totalSegmentsToBeDeleted;
+    // Total segments processed so far - across all replicas
+    @JsonProperty("totalRemainingSegmentsToBeAdded")
+    public int _totalRemainingSegmentsToBeAdded;
+    @JsonProperty("totalRemainingSegmentsToBeDeleted")
+    public int _totalRemainingSegmentsToBeDeleted;
+    @JsonProperty("totalRemainingSegmentsToConverge")
+    public int _totalRemainingSegmentsToConverge;
+    // Carry over stats - for when previous step's convergence doesn't complete and next step starts (bestEffort=true)
+    @JsonProperty("totalCarryOverSegmentsToBeAdded")
+    public int _totalCarryOverSegmentsToBeAdded;
+    @JsonProperty("totalCarryOverSegmentsToBeDeleted")
+    public int _totalCarryOverSegmentsToBeDeleted;
+    // Total new segments stats (not tracked by rebalance)
+    @JsonProperty("totalUniqueNewUntrackedSegmentsDuringRebalance")
+    public int _totalUniqueNewUntrackedSegmentsDuringRebalance;
+    // Derived stats
+    @JsonProperty("percentageTotalSegmentsAddsRemaining")
+    public double _percentageTotalSegmentsAddsRemaining;
```

Comment:
nit: _percentageTotalSegment`s`DeletesRemaining

maybe refine the two var names a bit, e.g. `percentRemainingSegmentsToBeAdded` to look closer with `totalRemainingSegmentsToBeAdded`

Reply:
done